agent/taskresource/grpcclient/credentialsfetcherclient.go

```diff
@@ -90,6 +93,71 @@ func (c CredentialsFetcherClient) AddKerberosLease(ctx context.Context, credenti
 	return credentialsFetcherResponse, nil
 }
 
+// AddNonDomainJoinedKerberosLease() invokes credentials fetcher daemon running on the host
+// to create kerberos tickets associated with gMSA accounts in domainless mode
+func (c CredentialsFetcherClient) AddNonDomainJoinedKerberosLease(ctx context.Context, credentialspecs []string, username string, password string, domain string) (CredentialsFetcherResponse, error) {
+	if len(credentialspecs) == 0 {
+		seelog.Error("credentialspecs request should not be empty")
+		return CredentialsFetcherResponse{}, nil
+	}
+
+	if len(username) == 0 || len(password) == 0 || len(domain) == 0 {
+		seelog.Error("username, password or domain should not be empty")
+		return CredentialsFetcherResponse{}, nil
+	}
+
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.CreateNonDomainJoinedKerberosLeaseRequest{CredspecContents: credentialspecs, Username: username, Password: password, Domain: domain}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.AddNonDomainJoinedKerberosLease(ctx, request)
+	if err != nil {
+		seelog.Errorf("could not create kerberos tickets: %v", err)
```

Comment:
Should we add more context to this log to identify the request? Preferably with `logger` package for structured logging (the structured logger has an `Error` field that we can use for `err`). 

Also, since we are returning the error, how about the consumer log the error? I see that the consumer of `AddKerberosLease` method logs the returned error so maybe we can follow the same pattern.

Same comment for `RenewNonDomainJoinedKerberosLease` method.